File: Server/src/main/core/cache/crypto/ISAACPair.java
package core.cache.crypto;

/**
 * Isaac pair.
 */
public final class ISAACPair {

    private ISAACCipher input;

    private ISAACCipher output;

    /**
     * Instantiates a new Isaac pair.
     *
     * @param input  the input
     * @param output the output
     */
    public ISAACPair(ISAACCipher input, ISAACCipher output) {
        this.input = input;
        this.output = output;
    }

    /**
     * Gets input.
     *
     * @return the input
     */
    public ISAACCipher getInput() {
        return input;
    }

    /**
     * Gets output.
     *
     * @return the output
     */
    public ISAACCipher getOutput() {
        return output;
    }

}

File: Emulator/src/main/core/cache/crypto/ISAACPair.kt
package core.cache.crypto

/**
 * Represents a ISAAC key pair, for both input and output.
 * @author `Discardedx2
 */
class ISAACPair
/**
 * Constructs a new `ISAACPair` `Object`.
 *
 * @param input  The input cipher.
 * @param output The output cipher.
 */(
    /**
     * The input cipher.
     */
    val input: ISAACCipher,
    /**
     * The output cipher.
     */
    @JvmField val output: ISAACCipher
)